cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/AnomalyDetails.java

```diff
@@ -75,6 +77,15 @@
           anomalyDetails.put(OPTIMIZATION_RESULT, goalViolations.optimizationResult(_isJson));
         }
         break;
+      case INTRA_BROKER_GOAL_VIOLATION:
+        IntraBrokerGoalViolations intraBrokerGoalViolations = (IntraBrokerGoalViolations) _anomalyState.anomaly();
```

Comment:
Maybe we can combine both cases once we have the abstract class?
```suggestion
      case GOAL_VIOLATION:
      case INTRA_BROKER_GOAL_VIOLATION:
        IntraBrokerGoalViolations intraBrokerGoalViolations = (BaseGoalViolations) _anomalyState.anomaly();
```

Reply:
Hey @zornhsu , returned a bit to this.
My small concern on this is that it's used in other parts of the application and I would avoid obfuscating the difference between the two, maybe reuse the logic but keep them separate. WDYT?

Reply:
Hi @ecojan, I think what zorn suggested makes more sense. Do you have an example that merging them would cause confusions?